src/bin/pg_autoctl/service_keeper.c

```diff
@@ -801,6 +807,9 @@ keeper_node_active(Keeper *keeper, bool doInit)
 		}
 	}
 
+	/* Finally establish a connection for notifications if none present */
+	(void) pgsql_listen(&(keeper->monitor.notificationClient), emptyChannelList);
+
```

Comment:
I believe we don't need that here, the `monitor_wait_for_state_change` in the beginning of the main loop reconnects if needed. We have a now spurious "Lost connection" warning that we should probably get rid of in `monitor_wait_for_state_change` that said, I believe it's now expected to have to establish a connection every once in a while from that point.